src/realm/table.cpp

```diff
@@ -1080,70 +1093,117 @@ void Table::set_table_type(Type table_type)
     }
 
     REALM_ASSERT_EX(table_type == Type::TopLevel || table_type == Type::Embedded, table_type);
-    set_embedded(table_type == Type::Embedded);
+    set_embedded(table_type == Type::Embedded, handle_backlinks);
 }
 
-void Table::set_embedded(bool embedded)
+void Table::set_embedded(bool embedded, bool handle_backlinks)
 {
-    if (Replication* repl = get_repl()) {
-        if (repl->get_history_type() == Replication::HistoryType::hist_SyncClient) {
-            throw std::logic_error(util::format("Cannot change '%1' to embedded when using Sync.", get_name()));
-        }
-    }
```

Comment:
We don't do similar validation for any of the other schema change operations that aren't supported when using sync and I don't really see any point in doing it here.